File: src/main/java/com/example/demo/controller/UserController.java
package com.example.demo.controller;

import com.example.demo.entity.BaseEntity;
import com.example.demo.entity.UserBean;
import com.example.demo.service.UserService;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import io.swagger.annotations.ApiParam;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;


@Api
@RestController
@RequestMapping("/user")
public class UserController {

    @Autowired
    private UserService userService;

    @RequestMapping(value = "/login", method = RequestMethod.GET)
    @ApiOperation(value = "登录")
    @ApiParam()
    private BaseEntity login(String phone, String password) {
        BaseEntity baseEntity = new BaseEntity();
        UserBean login = userService.login(phone, password);
        if (login != null) {
            baseEntity.setMessage("success");
            baseEntity.setData(login);
        } else {
            baseEntity.setCode(-1);
            baseEntity.setMessage("账号或密码错误");
        }

        return baseEntity;
    }

    @ResponseBody
    @RequestMapping(value = "/register", method = RequestMethod.POST)
    @ApiOperation(value = "注册")
    private BaseEntity register(String phone, String password) {
        BaseEntity baseEntity = new BaseEntity();
        UserBean user = userService.findUser(phone);
        if (user != null) {
            baseEntity.setCode(-1);
            baseEntity.setMessage("该账号已注册");
            return baseEntity;
        }
        userService.register(phone, password);
        baseEntity.setMessage("success");
        baseEntity.setData("注册成功！");
        return baseEntity;
    }
}

File: src/main/java/com/example/demo/controller/UserController.kt
package com.example.demo.controller

import com.example.demo.Utils.PhoneUtils.Companion.isMobile
import com.example.demo.entity.BaseEntity
import com.example.demo.service.UserService
import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import io.swagger.annotations.ApiParam
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.bind.annotation.RestController

/**
 * @Author Mr.Wu
 * @Date create on 2020/1/11
 * @DES
 */
@Api
@RestController
@RequestMapping("/user")
class UserController : BaseController() {

    @Autowired
    private val userService: UserService? = null

    /**
     * @param phone
     * @param password
     * @return
     */
    @RequestMapping(value = ["/login"])
    @ApiOperation(value = "登录")
    @ApiParam()
    private fun login(phone: String, password: String): BaseEntity? {
        return if (isMobile(phone)) {
            val login = userService?.login(phone, password)
            if (login != null)
                postData(data = login)
            else
                postData(code = -1, message = "账号或密码错误")

        } else
            postData(code = -2, message = "请输入正确的手机号")


    }

    /**
     * @param phone
     * @param password
     * @return
     */
    @ResponseBody
    @RequestMapping(value = ["/register"], method = [RequestMethod.POST])
    @ApiOperation(value = "注册")
    private fun register(phone: String, password: String): BaseEntity? {
        return if (isMobile(phone)) {
            val user = userService?.findUser(phone)
            if (user != null) {
                postData(code = -1, message = "该账号已注册")
            } else {
                userService?.register(phone, password)
                postData(message = "success")

            }
        } else {
            postData(code = -2, message = "请输入正确的手机号")
        }
    }

}